compose.yaml

```diff
@@ -0,0 +1,42 @@
+version: '3.8'
+
+services:
+  frontend:
+    build:
+      context: frontend
+      dockerfile: Dockerfile
+    networks:
+      - opendevin
+    ports:
+      - "127.0.0.1:3001:3001"
+    container_name: devin-frontend
+    depends_on:
+      - backend
+    volumes:
+      - ./config.toml:/usr/src/app/config.toml
+
+    command: ["npm", "start", "--", "--host", "0.0.0.0"]
```

Comment:
- I think maybe the vite server needs host 0.0.0.0 too, seemed like it on my similar docker-compose testing:
- https://github.com/OpenDevin/OpenDevin/blob/main/frontend/vite.config.ts#L17
```typescript
  server: {
    host: '0.0.0.0',
    port: 3001,
    proxy: {
      "/api": {
        target: `http://${BACKEND_HOST}/`,
        changeOrigin: true,
        rewrite: (path: string) => path.replace(/^\/api/, ""),
      },
      "/ws": {
        target: `ws://${BACKEND_HOST}/`,
        ws: true,
      },
    },
```

Reply:
I'm still optimizing here, but this is the only way I could get it to work so far

1. Change `BACKEND_HOST` to the backend docker dns name
```bash
    environment:
      - BACKEND_HOST=devin-backend:3000
```

2. Add exceptions to defineConfig
```
export default defineConfig({
  // depending on your application, base can also be "/"
  base: "",
  plugins: [react(), viteTsconfigPaths()],
  clearScreen: false,
  server: {
     port: process.env.FRONTEND_PORT
       ? Number.parseInt(process.env.FRONTEND_PORT, 10)
       : 3001,
     proxy: {
       "/api": {
         target: `http://${BACKEND_HOST}/`,
         changeOrigin: true,
         rewrite: (path: string) => path.replace(/^\/api/, ""),
       },
       "/ws": {
         target: `ws://${BACKEND_HOST}/`,
         ws: true,
       },
       "/default-model": {
         target: `http://${BACKEND_HOST}/`,
         changeOrigin: true,
         rewrite: (path: string) => path.replace(/^\/default-model/, ""),
       },
       "/litellm-agents": {
         target: `http://${BACKEND_HOST}/`,
         changeOrigin: true,
         rewrite: (path: string) => path.replace(/^\/litellm-agents/, ""),
       },
     },
  },
 });
 ```
 
 I think it might be possible to do this without the changes to the config.ts, but I haven't found it yet.